File: qqbot-protocol/src/main/java/com/mieai/qqbot/protocol/gateway/GatewayHello.java
package com.mieai.qqbot.protocol.gateway;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/** Data carried by a Gateway Hello envelope. */
@JsonIgnoreProperties(ignoreUnknown = true)
public record GatewayHello(@JsonProperty("heartbeat_interval") long heartbeatInterval) {}

File: qqbot-protocol/src/main/kotlin/com/mieai/qqbot/protocol/gateway/GatewayHello.kt
package com.mieai.qqbot.protocol.gateway

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/** Data carried by a Gateway Hello envelope. */
@JsonIgnoreProperties(ignoreUnknown = true)
data class GatewayHello(@JsonProperty("heartbeat_interval") val heartbeatInterval: Long)
